experiment/pip2build.py

```diff
@@ -0,0 +1,173 @@
+        if tver:
+            ver = tver
+        if ver and pkg:
+            return pkg, ver
+    raise ValueError('Could not find Name and Version', pkg, ver, pkg_info)
+
+
+
+def convert(session, full, pkgs):  # pylint: disable=too-many-statements,too-many-locals,too-many-branches
+    base = os.path.basename(full)
+    stderr('%s: converting...', base)
+    deps = []
+    with tarfile.open(full) as tar:
+        for name in tar.getnames():
+            if 'PKG-INFO' in name:
+                fp = tar.extractfile(name)
+                try:
+                    pkg, ver = extract_info(fp)
+                finally:
+                    fp.close()
+                break
+        else:
+            raise ValueError('PKG-INFO not in %s', base)
+        stderr('  info: %s %s', pkg, ver)
+        try:
+            root = '%s-%s/%s' % (pkg, ver, pkg)
+            tar.getmember(root)
+        except KeyError:
+            root = '%s-%s' % (pkg, ver)
+            tar.getmember(root)
+        try:
+            mem = tar.getmember('%s-%s/%s.egg-info/requires.txt' % (pkg, ver, pkg))
+        except KeyError:
+            mem = None
+        if mem:
+            fp = tar.extractfile(mem)
+            try:
+                for line in fp:
+                    # yes
+                    mat = re.search(r'^([\w._-]+)', line)
+                    if not mat:
+                        continue
+                    dep = mat.group(1)
+                    for option in pkgs:
+                        if not option.startswith('%s-' % dep.replace('-', '_')):
+                            continue
+                        deps.append(dep)
+            finally:
+                fp.close()
+        if deps:
+            stderr('  deps: %s', ', '.join(deps))
+        else:
+            stderr('  no deps')
+
+    pkg_url = 'https://pypi.python.org/pypi/%s/json' % pkg
+    resp = session.get(pkg_url)
+    resp.raise_for_status()
+    meta = json.loads(resp.content)
+    for option in meta['releases'][ver]:
+        if option['python_version'] == 'source' and option['url'].endswith('.tar.gz'):
+            url = option['url']
+            md5 = option['md5_digest']
+            break
+    else:
+        raise ValueError('Cannot find source url for %s' % pkg)
+    stderr('  url: %s', url)
+    with open(full) as fp:
+        buf = fp.read()
+    digest = hashlib.md5(buf).hexdigest()
+    if digest != md5:
+        raise ValueError('Downloaded md5: %s != expected: %s' % (digest, md5))
+
+    sha = hashlib.sha256()
+    with open(full) as fp:
+        buf = True
+        while buf:
+            buf = fp.read(10**6)
+            sha.update(buf)
+    digest = sha.hexdigest()
+    with open('BUILD-%s' % pkg, 'w') as fp:
+        dep_lines = deps and [
+            '    deps = [',
+            '\n'.join('      "@%s//:%s",' % (d, d) for d in deps),
+            '    ],',
+        ] or []
+        fp.write('\n'.join([
+            '',
+            'new_http_archive(',
```

Comment:
new_http_archive is a WORKSPACE rule, not a BUILD rule, right? am I misunderstanding?